lib/ansible/plugins/action/nxos.py

```diff
@@ -43,8 +43,18 @@ def run(self, tmp=None, task_vars=None):
 
         socket_path = None
 
+        if self._task.action == 'nxos_file_copy':
+            if self._play_context.connection == 'network_cli':
```

Comment:
`host` and `password` don't change, I would just set them all as if using network_cli, then `if connection == local: username = connection_user`. This has the side benefit of being really easy to clean up when local goes away

Reply:
@Qalthos Of course. I just did that to keep the blocks distinct. But I don't have strong opinion on this. Makes sense and pushed.